### PR TITLE
Allow configuring and building on Win with MSVC and nonVS generators

### PR DESCRIPTION
**Pull request by jpfeuffer**
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Any generator that is available for win. Nmake or Ninja for example.
For building in VS Code and CLion etc.
Not the bloated full VS.

File: doc/doxygen/install/install-win.doxygen [hunk 1]
```diff
@@ -78,22 +79,23 @@
 
     <ol>
       <li>Open a <b>VisualStudio Developer Command prompt</b> (by default you will get a 32-bit environment which you most likely do not want; use a x64 environment if you want 64 bit apps!)<br>
-					@note VS2015 does not provide a commandline shortcut for a 64bit command prompt. Call <tt>"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64</tt> to get one!
-					@note VS2017 has a (well hidden) batch-file which correctly configures your environment for 64bit builds. It should be located at
-					<tt>"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars64.bat"</tt>. (Same goes for VS2019)
+					@note VS2017+ has a (well hidden) batch-file which correctly configures your environment for 64bit builds. It should be located at
```
Comment (cbielow) [inline doc/doxygen/install/install-win.doxygen:82]:
```suggestion
					@note VS2019 and above have a (well hidden) batch-file which correctly configures your environment for 64bit builds. It should be located at
```

File: doc/doxygen/install/install-win.doxygen [hunk 2]
```diff
@@ -32,11 +32,12 @@
...
+      <li><b>Microsoft Visual C++ Compiler</b>: Comes with <a href="https://visualstudio.microsoft.com/de/downloads/?q=build+tools">Microsoft Visual Studio Build Tools 2019 or above </a>.
+        Scroll to the bottom of the page to get the build tools only. If you want, you can also download the full Visual Studio (Community/Pro/Enterprise) IDE.
+        <br><b>Not supported (or at least not tested) is MinGW (g++ based compiler). Do not use a MinGW-enabled shell since CMake will get confused! Please use the Command Prompt of the Visual Studio version you want to build with or load vcvarsall.bat.</b>
             Other compilers which are close to the VS toolchain might work, but are not tested.
       <li><b>CMake</b>: Have at least CMake 3.14+ or greater installed (see http://www.cmake.org). <br>
-	      CMake's Generator (-G flag) also must support your version of Visual Studio. If it doesn't, please upgrade CMake.<br>
+	      If you are using Visual Studio 2019+ as IDE, make sure CMake's Generator (-G flag) supports your version of Visual Studio. If it doesn't, please upgrade CMake.<br>
```
Comment (cbielow) [inline doc/doxygen/install/install-win.doxygen:40]:
```suggestion
	      If you are using the Visual Studio IDE, make sure CMake's Generator (-G flag) supports your version of Visual Studio. If it doesn't, please upgrade CMake.<br>
```

File: doc/doxygen/install/install-win.doxygen [hunk 3]
```diff
@@ -313,6 +315,33 @@
...
+      <li> Ctrl+Shift+P and find <code>Git: clone</code>, then type and search for OpenMS/OpenMS. Select a folder. Wait for clone to finish. Open clone. Trust authors.
+      <li> Open bash Terminal from within VS Code and type <code>source tools/loadvcvarsall.sh && load_msenv && git submodule update --init contrib && mkdir contrib-build && cd contrib-build && cmake -DBUILD_TYPE=ALL ../contrib</code>
+      <li> Adapt (to your paths) and add
+        \code
+        "cmake.configureArgs": [
+          "-DOPENMS_CONTRIB_LIBS=C:\\git\\contrib-build",
+          "-DCMAKE_PREFIX_PATH=C:\\dev\\Qt\\5.15.2\\msvc2019_64"
+        ]
+        \endcode
+        to your settings or configure manually.
```
Comment (cbielow) [inline doc/doxygen/install/install-win.doxygen:335]: Where would I put this in VS Code for example?

File: tools/load_vcvarsall.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            msversion_prod=15
+            ;;
+        2019)
+            msversion_prod=16
+            ;;
+        2022)
+            msversion_prod=17
+            ;;
+        *)
+            >&2 printf "Invalid version year. Supported are 2017, 2019 and 2022."
```
Comment (cbielow) [inline tools/load_vcvarsall.sh:22]: do we still support VS 2017? It does not (fully) support C17, right?

File: tools/load_vcvarsall.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            return 1
+    esac
+
+    msversion_prod_p1=$(($msversion_prod+1))
+
+    local VSWHERE='C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe'
+    installPath=$("$VSWHERE" -products '*' -version "[$msversion_prod,$msversion_prod_p1)" -property installationPath)
+
+    local vcvarsall="${installPath}\\VC\\Auxiliary\\Build\\vcvarsall.bat"
```
Comment (cbielow) [inline tools/load_vcvarsall.sh:31]: check if it exists? print error otherwise?

**Review by cbielow — COMMENTED** [review]
looks good.
Only minor stuff

**Comment by cbielow** [comment]
> Any generator that is available for win. Nmake or Ninja for example. For building in VS Code and CLion etc. Not the bloated full VS.

Maybe add that information somewhere at the top?!

File: doc/doxygen/install/install-win.doxygen (hunk 3 above)
Comment (jpfeuffer) [reply doc/doxygen/install/install-win.doxygen:335]: oh in your user settings

File: tools/load_vcvarsall.sh (hunk 4 above)
Comment (jpfeuffer) [reply tools/load_vcvarsall.sh:22]: I think not, yes

File: doc/doxygen/install/install-win.doxygen (hunk 3 above)
Comment (jpfeuffer) [inline doc/doxygen/install/install-win.doxygen:335]:
```suggestion
        to your settings or configure manually in your user-config.yml or via the VS Code command runner.
```

File: tools/load_vcvarsall.sh (hunk 4 above)
Comment (jpfeuffer) [reply tools/load_vcvarsall.sh:22]: but the script might be helpful for older versions as well.

**Pull request merged**